tests/devices/qubit/test_adjoint_jacobian.py

```diff
@@ -282,3 +282,191 @@ def test_gradient_of_tape_with_tensor(self, tol):
+
+
+class TestAdjointJVP:
+    """Test for adjoint_jvp"""
+
+    @pytest.mark.parametrize("tangents", [(0,), (1.232,)])
+    def test_single_param_single_obs(self, tangents, tol):
+        """Test JVP is correct for a single parameter and observable"""
+        x = np.array(0.654)
+        qs = QuantumScript([qml.RY(x, 0)], [qml.expval(qml.PauliZ(0))])
+        qs.trainable_params = {0}
+
+        actual = adjoint_jvp(qs, tangents)
+        assert isinstance(actual, np.ndarray)
+
+        jac = adjoint_jacobian(qs)
+        expected = jac * tangents[0]
+        assert np.allclose(actual, expected, atol=tol)
+
+    @pytest.mark.parametrize("tangents", [(0,), (1.232,)])
+    def test_single_param_multi_obs(self, tangents, tol):
+        """Test JVP is correct for a single parameter and multiple observables"""
+        x = np.array(0.654)
+        qs = QuantumScript([qml.RY(x, 0)], [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(0))])
+        qs.trainable_params = {0}
+
+        actual = adjoint_jvp(qs, tangents)
+        assert isinstance(actual, tuple)
+        assert len(actual) == 2
+        assert all(isinstance(r, np.ndarray) for r in actual)
+
+        jac = adjoint_jacobian(qs)
+        expected = np.array(jac) * tangents[0]
+        assert np.allclose(actual, expected, atol=tol)
+
+    @pytest.mark.parametrize("tangents", [(0, 0), (0, 0.653), (1.232, 2.963)])
+    def test_multi_param_single_obs(self, tangents, tol):
+        """Test JVP is correct for multiple parameters and a single observable"""
+        x = np.array(0.654)
+        y = np.array(1.221)
+
+        qs = QuantumScript([qml.RY(x, 0), qml.RZ(y, 0)], [qml.expval(qml.PauliY(0))])
+        qs.trainable_params = {0, 1}
+
+        actual = adjoint_jvp(qs, tangents)
+        assert isinstance(actual, np.ndarray)
+
+        jac = adjoint_jacobian(qs)
+        expected = np.dot(np.array(jac), np.array(tangents))
+        assert np.allclose(actual, expected, atol=tol)
+
+    @pytest.mark.parametrize("tangents", [(0, 0), (0, 0.653), (1.232, 2.963)])
+    def test_multi_param_multi_obs(self, tangents, tol):
+        """Test JVP is correct for multiple parameters and observables"""
+        x = np.array(0.654)
+        y = np.array(1.221)
+
+        obs = [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(0)), qml.expval(qml.PauliY(0))]
+        qs = QuantumScript([qml.RY(x, 0), qml.RZ(y, 0)], obs)
+        qs.trainable_params = {0, 1}
+
+        actual = adjoint_jvp(qs, tangents)
+        assert isinstance(actual, tuple)
+        assert len(actual) == 3
+        assert all(isinstance(r, np.ndarray) for r in actual)
+
+        jac = np.array(adjoint_jacobian(qs))
+        expected = jac @ np.array(tangents)
+        assert np.allclose(actual, expected, atol=tol)
+
+    @pytest.mark.parametrize("tangents", [(0, 0), (0, 0.653), (1.232, 2.963)])
+    def test_custom_wire_labels(self, tangents, tol):
+        """Test JVP is correct for custom wire labels"""
```

Comment:
related to my concern above, could you add a test with out-of-order wire labels (that when sorted appear to be standard, like `[2, 0, 1]`) to ensure that works as expected?

Reply:
Sure, I can add a test here.